decoder.go

```diff
@@ -4,14 +4,64 @@ import (
 	"fmt"
 	"image"
 	"unsafe"
+
+	"github.com/pkg/errors"
 )
 
-// #cgo pkg-config: libavcodec libavutil libswscale
-// #include <libavcodec/avcodec.h>
-// #include <libavutil/imgutils.h>
-// #include <libswscale/swscale.h>
+/*
+#cgo pkg-config: libavcodec libavutil libswscale libavformat
+#include <libavcodec/avcodec.h>
+#include <libavutil/imgutils.h>
+#include <libswscale/swscale.h>
+#include <libavformat/avformat.h>
+#include <libavutil/avutil.h>
+#include <stdlib.h>
+
+int get_video_codec(AVFormatContext *avFormatCtx) {
```

Comment:
please add a comment here for what this function does.

Reply:
Sounds good, added comment to helper.